### PR TITLE
overarch 0.14.0 (new formula)

### PR DESCRIPTION
**Pull request by sgerrand**
💁 This change adds a formula for [overarch](https://github.com/soulspace-org/overarch), a generator of data driven descriptions for software architecture based on UML and the C4 model.

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/o/overarch.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  sha256 "7ba86645fa2cf9893a9c9c2eda416d4f4ca0b2bcbde683b4b7b0fc72b9bca909"
+  license "EPL-1.0"
+  head "https://github.com/soulspace-org/overarch.git", branch: "main"
+
+  depends_on "leiningen" => :build
+  depends_on "openjdk"
+
+  resource "jar" do
+    url "https://github.com/soulspace-org/overarch/releases/download/v0.14.0/overarch.jar"
+    sha256 "7d573e0c044e63487cf9104142651e79beb849282916cf63dcb3a3256211bff0"
```
Comment (SMillerDev) [inline Formula/o/overarch.rb:14]: Why not use this as the URL?

Comment (sgerrand) [reply Formula/o/overarch.rb:14]: I assume you mean why not lift that `url` out of the `jar` block into the main definition. That would preclude being able to build this formula from source via `lein uberjar`, which is what I was thinking of adding next.

Comment (sgerrand) [reply Formula/o/overarch.rb:14]: Do you have any thoughts on that?

Comment (SMillerDev) [reply Formula/o/overarch.rb:14]: Then what is the jar for? I'm confused, that seems like a duplicate install

Comment (sgerrand) [reply Formula/o/overarch.rb:17]: See 138938139e6 for changes to support building this formula from upstream's HEAD commit.

Comment (sgerrand) [reply Formula/o/overarch.rb:17]: I hope that makes more sense now. Let me know if you have any further feedback or need clarification.

Comment (sgerrand) [reply Formula/o/overarch.rb:17]: 🤦 Ah, I see what you mean now. Sorry for missing that earlier. I'll resolve it now.

Comment (sgerrand) [reply Formula/o/overarch.rb]: Should be okay now.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @sgerrand!

**Comment by chenrui333** [comment]
your base branch is a bit outdated.

```
  * The following should either be an alias or a rename, not both: bigdata, boot2docker, camlistore, crystal-lang, findbugs, gitlab-ci-multi-runner, gnome-icon-theme, gnupg2, gtef, gtk+4, htop-osx, latexila, letsencrypt, libmongoclient, linux-headers, mat, mobile-shell, mongo-c, newsbeuter, osh, pyqt5, qt5, recipes, saltstack, speedtest_cli, ssreflect, tachyon, tensorflow and transfig
Error: 1 problem in 1 tap detected.
```

**Review by chenrui333 — APPROVED** [review]

**Pull request merged**